### PR TITLE
Remove usage of CoreLocation when possible

### PR DESCRIPTION
**Pull request by Odrakir**
This was already half done, so I'm going all the way before the integration with CM (which uses Location a lot more).

Basically is removing `import CoreLocation` when possible and replacing `CLLocation` instances by our own `Location` instances.

It's a lot of files, but basically is replacing `CLLocation` with `Location`.

### CODE REVIEW

File: teferiTests/Pipeline/PersistencySinkTests.swift [hunk 1]
```diff
@@ -63,16 +62,25 @@ class PersistencySinkTests : XCTestCase
...
+                                     latitude: 38.628060, longitude: -117.848463,
+                                     speed: 0, course: 0, altitude: 0,
+                                     verticalAccuracy: 0, horizontalAccuracy: 0)
         
-        data[4] = data[4].with(location: Location(fromCLLocation: CLLocation(latitude: 38.628060, longitude: -117.848463)))
-        data[5] = data[5].with(location: Location(fromCLLocation: expectedLocation))
+        let expectedLocation = Location(timestamp: Date(),
+                                        latitude: 37.628060, longitude: -116.848463,
+                                        speed: 0, course: 0, altitude: 0,
+                                        verticalAccuracy: 0, horizontalAccuracy: 0)
```
Comment (sofokli1) [inline teferiTests/Pipeline/PersistencySinkTests.swift]: if not default values, maybe an extension in the test target to remove those `0`?

File: teferi/Services/Implementations/Persistency/Adapters/CoreDataModelAdapter.swift [hunk 2]
```diff
@@ -16,18 +15,20 @@ class CoreDataModelAdapter<T>
...
         let possibleTime = managedObject.value(forKey: timeKey) as? Date
         let possibleLatitude = managedObject.value(forKey: latKey) as? Double
         let possibleLongitude = managedObject.value(forKey: lngKey) as? Double
         
         if let time = possibleTime, let latitude = possibleLatitude, let longitude = possibleLongitude
         {
-            let coord = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
-            location = CLLocation(coordinate: coord, altitude: 0, horizontalAccuracy: 0, verticalAccuracy: 0, timestamp: time)
+            location = Location(timestamp: time,
+                                latitude: latitude, longitude: longitude,
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply teferi/Services/Implementations/Persistency/Adapters/CoreDataModelAdapter.swift]: I'll put latitude and longitude in the same line, being related.

File: teferiTests/Helpers/LocationExtensions.swift [hunk 3]
```diff
@@ -46,72 +48,63 @@ extension CLLocationCoordinate2D
...
-                                     altitude: altitude,
-                                     horizontalAccuracy: horizontalAccuracy,
-                                     verticalAccuracy: verticalAccuracy,
-                                     timestamp: timestamp ?? self.timestamp)
+        let newCoordinates = offsetCoordinates(direction, meters: meters)
+        let newLocation = Location(timestamp: timestamp ?? self.timestamp,
+                                   latitude: newCoordinates.latitude, longitude: newCoordinates.longitude,
+                                   speed: 0, course: 0, altitude: altitude,
+                                   verticalAccuracy: horizontalAccuracy, horizontalAccuracy: horizontalAccuracy)
+
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply teferiTests/Helpers/LocationExtensions.swift]: Related params in the same line, I don't want to make this too height unnecessarily.

**Comment by Odrakir** [comment]
I swear in my head this makes sense. `Location` should be a struct, not a class (as should every other model), and then it'll have a default initializer with all properties. Then we'll only need an extension with whichever convenience initializers we need. I don't want to make more changes in this PR, so more of this soon.

**Pull request merged**